checker/jdk/index/src/java/util/Calendar.java

```diff
@@ -0,0 +1,2769 @@
+                shortNames.putAll(longNames);
+            }
+            return shortNames;
+        }
+
+        // SHORT or LONG
+        return getDisplayNamesImpl(field, style, locale);
+    }
+
+    private Map<String, Integer> getDisplayNamesImpl(int field, int style, Locale locale) {
+        DateFormatSymbols symbols = DateFormatSymbols.getInstance(locale);
+        String[] strings = getFieldStrings(field, style, symbols);
+        if (strings != null) {
+            Map<String, Integer> names = new HashMap<String, Integer>();
+            for (int i = 0; i < strings.length; i++) {
+                if (strings[i].length() == 0) {
+                    continue;
+                }
+                names.put(strings[i], i);
+            }
+            return names;
+        }
+        return null;
+    }
+
+    boolean checkDisplayNameParams(int field, int style, int minStyle, int maxStyle,
+                                   Locale locale, int fieldMask) {
+        if (field < 0 || field >= fields.length ||
+                style < minStyle || style > maxStyle) {
+            throw new IllegalArgumentException();
+        }
+        if (locale == null) {
+            throw new NullPointerException();
+        }
+        return isFieldSet(fieldMask, field);
+    }
+
+    private String[] getFieldStrings(int field, int style, DateFormatSymbols symbols) {
+        String[] strings = null;
+        switch (field) {
+            case ERA:
+                strings = symbols.getEras();
+                break;
+
+            case MONTH:
+                strings = (style == LONG) ? symbols.getMonths() : symbols.getShortMonths();
+                break;
+
+            case DAY_OF_WEEK:
+                strings = (style == LONG) ? symbols.getWeekdays() : symbols.getShortWeekdays();
+                break;
+
+            case AM_PM:
+                strings = symbols.getAmPmStrings();
+                break;
+        }
+        return strings;
+    }
+
+    /**
+     * Fills in any unset fields in the calendar fields. First, the {@link
+     * #computeTime()} method is called if the time value (millisecond offset
+     * from the <a href="#Epoch">Epoch</a>) has not been calculated from
+     * calendar field values. Then, the {@link #computeFields()} method is
+     * called to calculate all calendar field values.
+     */
+    protected void complete() {
+        if (!isTimeSet)
+            updateTime();
+        if (!areFieldsSet || !areAllFieldsSet) {
+            computeFields(); // fills in unset fields
+            areAllFieldsSet = areFieldsSet = true;
+        }
+    }
+
+    /**
+     * Returns whether the value of the specified calendar field has been set
+     * externally by calling one of the setter methods rather than by the
+     * internal time calculation.
+     *
+     * @return <code>true</code> if the field has been set externally,
+     * <code>false</code> otherwise.
+     * @throws IndexOutOfBoundsEx
```

Comment:
Could be `@IntRange(from = 1, to = 7) int value`. Also `setMinimalDaysInFirstWeek` below.